### PR TITLE
various ros2-nav2 changes

### PR DESCRIPTION
**Pull request by girvenavery2022**
## changes made 

# slam toolbox 
1. adjusted map update intervals as well as tf2 buffer and stack size
2. adjusted scanmatcher travel_heading, travel_distance, scan buffer size/link
3. adjusted loop closure and correlation parameters

# Nav2 
1. replaced DWB with regulated pure pursuit 
2. corrected local costmap to use rectangle robot footprint
3. increased the radius on global costmap robot footprint
4. update map server to use a yaml file from us (only if we use amcl)
5. I hate dijkstra's algorithm with a burning passion so Nav2 now uses A*

# mammoth launch file 
1. added sim time to nodes 

# mammoth sdf world
1. added a cafe from ignition gazebo

### CODE REVIEW

File: mammoth_gazebo/config/nav2_params.yaml [hunk 1]
```diff
@@ -238,7 +222,9 @@ global_costmap:
 map_server:
   ros__parameters:
     use_sim_time: True
-    yaml_filename: "turtlebot3_world.yaml"
+    yaml_filename: "/ros2_nav_ws/map_1636071052.yaml"
```
[the remark this answers is not in the document]
Comment (dcutting133) [reply mammoth_gazebo/config/nav2_params.yaml:225]: bump

**Review by dcutting133 — CHANGES_REQUESTED** [review]

File: mammoth_gazebo/launch/include/navigation.launch.py [hunk 2]
```diff
@@ -85,7 +84,7 @@ def generate_launch_description():
 
         # Nodes
         slam_toolbox,
-        # nav2_stack,  # 5/23/21 dcutting133: currently this is very buggy.
+        #nav2_stack,  # 5/23/21 dcutting133: currently this is very buggy.
```
Comment (dcutting133) [inline mammoth_gazebo/launch/include/navigation.launch.py:87]: uncomment and use the correct launch file for what is being launched by you?

File: mammoth_gazebo/config/nav2_params.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (girvenavery2022) [reply mammoth_gazebo/config/nav2_params.yaml]: fixed in my latest commit:)

File: mammoth_gazebo/launch/include/navigation.launch.py (hunk 2 above)
Comment (girvenavery2022) [reply mammoth_gazebo/launch/include/navigation.launch.py]: fixed in my latest commit:) I had to make a nav2 launch file from scratch for it to work

**Review by dcutting133 — APPROVED** [review]

**Pull request merged**